velox/dwio/parquet/tests/reader/ParquetReaderBenchmarkMain.cpp

```diff
@@ -0,0 +1,2325 @@
+      50000,                                                              \
+      true);                                                              \
+  BENCHMARK_NAMED_PARAM(                                                  \
+      run,                                                                \
+      _name_##_Filter_##_filter_##_Nulls_##_null_##_next_100k_dict,       \
+      #_name_,                                                            \
+      _type_,                                                             \
+      _filter_,                                                           \
+      _null_,                                                             \
+      100000,                                                             \
+      false);                                                             \
+  BENCHMARK_NAMED_PARAM(                                                  \
+      run,                                                                \
+      _name_##_Filter_##_filter_##_Nulls_##_null_##_next_100k_plain,      \
+      #_name_,                                                            \
+      _type_,                                                             \
+      _filter_,                                                           \
+      _null_,                                                             \
+      100000,                                                             \
+      true);                                                              \
+  BENCHMARK_DRAW_LINE();
+
+#define PARQUET_BENCHMARKS_FILTERS(_type_, _name_, _filter_)    \
+  PARQUET_BENCHMARKS_FILTER_NULLS(_type_, _name_, _filter_, 0)  \
+  PARQUET_BENCHMARKS_FILTER_NULLS(_type_, _name_, _filter_, 20) \
+  PARQUET_BENCHMARKS_FILTER_NULLS(_type_, _name_, _filter_, 50) \
+  PARQUET_BENCHMARKS_FILTER_NULLS(_type_, _name_, _filter_, 70) \
+  PARQUET_BENCHMARKS_FILTER_NULLS(_type_, _name_, _filter_, 100)
+
+#define PARQUET_BENCHMARKS(_type_, _name_)        \
+  PARQUET_BENCHMARKS_FILTERS(_type_, _name_, 0)   \
+  PARQUET_BENCHMARKS_FILTERS(_type_, _name_, 20)  \
+  PARQUET_BENCHMARKS_FILTERS(_type_, _name_, 50)  \
+  PARQUET_BENCHMARKS_FILTERS(_type_, _name_, 70)  \
+  PARQUET_BENCHMARKS_FILTERS(_type_, _name_, 100) \
+  BENCHMARK_DRAW_LINE();
+
+#define PARQUET_BENCHMARKS_NO_FILTER(_type_, _name_) \
+  PARQUET_BENCHMARKS_FILTERS(_type_, _name_, 100)    \
+  BENCHMARK_DRAW_LINE();
+
+PARQUET_BENCHMARKS(DECIMAL(18, 3), ShortDecimalType);
+PARQUET_BENCHMARKS(DECIMAL(38, 3), LongDecimalType);
+PARQUET_BENCHMARKS(VARCHAR(), Varchar);
+
+PARQUET_BENCHMARKS(BIGINT(), BigInt);
+PARQUET_BENCHMARKS(DOUBLE(), Double);
+PARQUET_BENCHMARKS_NO_FILTER(MAP(BIGINT(), BIGINT()), Map);
+PARQUET_BENCHMARKS_NO_FILTER(ARRAY(BIGINT()), List);
+
+// TODO: Add all data types
+
+int main(int argc, char** argv) {
+  folly::Init init{&argc, &argv};
+  memory::MemoryManager::initialize({});
+  folly::runBenchmarks();
+  return 0;
+}
+
+/*
```

Comment:
We can leave these numbers in commit message.  They are associated to a certain point in time (i.e. commit) so it is the best place it put them.

Reply:
That will make the commit message too long. I will add it in the PR message.